src/mlpack/methods/linear_regression/linear_regression.cpp

```diff
@@ -95,7 +100,7 @@ void LinearRegression::Predict(const arma::mat& points,
   {
     // We want to be sure we have the correct number of dimensions in the
     // dataset.
-    Log::Assert(points.n_rows == parameters.n_rows - 1);
+    util::CheckSameDimensionality(points, (unsigned long int) (parameters.n_rows-1), "LinearRegression::Predict()", "points");
```

Comment:
```suggestion
    util::CheckSameDimensionality(points, (size_t) (parameters.n_rows-1), "LinearRegression::Predict()", "points");
```
(I think `size_t` is the type used in those function definitions, so it's probably shorter to use that here too.)